packages/shorebird_cli/lib/src/commands/release_new/release_new_command.dart

```diff
@@ -0,0 +1,420 @@
+        );
+
+        // Ask the user to proceed (this is skipped when running via CI).
+        await confirmCreateRelease(
+          app: app,
+          releaseVersion: releaseVersion,
+          flutterVersion: targetFlutterRevision,
+          releasePlatform: releaser.releaseType.releasePlatform,
+        );
+
+        final release = await getOrCreateRelease(
+          version: releaseVersion,
+          releasePlatform: releaser.releaseType.releasePlatform,
+        );
+        await prepareRelease(release: release, pipeline: releaser);
+        await releaser.uploadReleaseArtifacts(release: release, appId: appId);
+        await finalizeRelease(release: release, pipeline: releaser);
+
+        logger
+          ..success('✅ Published Release ${release.version}!')
+          ..info(releaser.postReleaseInstructions);
+
+        printPatchInstructions(
+          releaser: releaser,
+          releaseVersion: releaseVersion,
+          releaseType: releaser.releaseType,
+          flavor: flavor,
+          target: target,
+        );
+      },
+      values: {
+        shorebirdEnvRef.overrideWith(() => releaseFlutterShorebirdEnv),
+      },
+    );
+  }
+
+  /// Determines which Flutter version to use for the release. This will be
+  /// either the version specified by the user or the version provided by
+  /// [shorebirdEnv]. Will exit with code 70 if the version specified by the
+  /// user is not found/supported.
+  Future<String> resolveTargetFlutterRevision() async {
+    if (flutterVersionArg != null) {
+      final String? revision;
+      try {
+        revision = await shorebirdFlutter.getRevisionForVersion(
+          flutterVersionArg!,
+        );
+      } catch (error) {
+        logger.err(
+          '''
+Unable to determine revision for Flutter version: $flutterVersionArg.
+$error''',
+        );
+        exit(ExitCode.software.code);
+      }
+
+      if (revision == null) {
+        final openIssueLink = link(
+          uri: Uri.parse(
+            'https://github.com/shorebirdtech/shorebird/issues/new?assignees=&labels=feature&projects=&template=feature_request.md&title=feat%3A+',
+          ),
+          message: 'open an issue',
+        );
+        logger.err(
+          '''
+Version $flutterVersionArg not found. Please $openIssueLink to request a new version.
+Use `shorebird flutter versions list` to list available versions.
+''',
+        );
+        exit(ExitCode.software.code);
+      }
+
+      return revision;
+    }
+
+    return shorebirdEnv.flutterRevision;
+  }
+
+  /// Asserts that a release with version [version] can be released using
+  /// flutter revision [flutterRevision]. If a release has already been
+  /// published with the given [version] for the platform [releasePlatform], or
+  /// if a release already exists with [version] but was compiled with a
+  /// different Flutter revision, an error will be thrown.
+  Future<void> assertVersionIsReleasable({
```

Comment:
```suggestion
  Future<void> ensureVersionIsReleasable({
```
for consistency with `codePushClientWrapper.ensureReleaseIsNotActive`

Reply:
Done